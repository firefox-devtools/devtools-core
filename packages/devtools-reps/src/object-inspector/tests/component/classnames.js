/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

const { mount } = require("enzyme");
const React = require("react");
const { createFactory } = React;
const ObjectInspector = createFactory(require("../../index"));
const ObjectClient = require("../__mocks__/object-client");

function generateDefaults(overrides) {
  return Object.assign({
    autoExpandDepth: 0,
    roots: [{
      path: "root",
      name: "root",
      contents: {value: 42}
    }],
    createObjectClient: grip => ObjectClient(grip)
  }, overrides);
}

describe("ObjectInspector - classnames", () => {
  it("has the expected class", () => {
    const props = generateDefaults();
    const wrapper = mount(ObjectInspector(props));

    expect(wrapper.hasClass("tree")).toBeTruthy();
    expect(wrapper.hasClass("inline")).toBeFalsy();
    expect(wrapper.hasClass("nowrap")).toBeFalsy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("has the nowrap class when disableWrap prop is true", () => {
    const props = generateDefaults({
      disableWrap: true,
    });
    const wrapper = mount(ObjectInspector(props));
    expect(wrapper.hasClass("nowrap")).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("has the inline class when inline prop is true", () => {
    const props = generateDefaults({
      inline: true,
    });
    const wrapper = mount(ObjectInspector(props));
    expect(wrapper.hasClass("inline")).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
