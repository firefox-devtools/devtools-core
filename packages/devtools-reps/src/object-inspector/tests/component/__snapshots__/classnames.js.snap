// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectInspector - classnames has the expected class 1`] = `
"<ObjectInspector autoExpandDepth={0} roots={{...}} createObjectClient={[Function]}>
  <Tree className=\\"object-inspector\\" autoExpandAll={true} autoExpandDepth={0} disabledFocus={[undefined]} itemHeight={20} isExpanded={[Function]} isExpandable={[Function]} focused={{...}} getRoots={[Function]} getParent={[Function]} getChildren={[Function]} getKey={[Function]} onExpand={[Function]} onCollapse={[Function]} onFocus={[Function]} renderItem={[Function]}>
    <div className=\\"tree object-inspector\\" role=\\"tree\\" tabIndex=\\"0\\" onKeyDown={[Function]} onKeyPress={[Function]} onKeyUp={[Function]} onFocus={[Function]} onBlur={[Function]} onClick={[Function]} aria-label={[undefined]} aria-labelledby={[undefined]} aria-activedescendant={{...}} style={{...}}>
      <TreeNode id=\\"root\\" index={0} item={{...}} depth={0} renderItem={[Function]} focused={false} expanded={false} isExpandable={false} onExpand={[Function]} onCollapse={[Function]} onClick={[Function]}>
        <div id=\\"root\\" className=\\"tree-node\\" style={{...}} onClick={[Function]} role=\\"treeitem\\" aria-level={0} aria-expanded={[undefined]} data-expandable={false}>
          <div className=\\"node object-node\\" onClick={[Function]} onDoubleClick={{...}}>
            <span className=\\"object-label\\" onClick={{...}}>
              root
            </span>
            <span className=\\"object-delimiter\\">
              : 
            </span>
            <span className=\\"objectBox objectBox-number\\">
              42
            </span>
          </div>
        </div>
      </TreeNode>
    </div>
  </Tree>
</ObjectInspector>"
`;

exports[`ObjectInspector - classnames has the inline class when inline prop is true 1`] = `
"<ObjectInspector autoExpandDepth={0} roots={{...}} createObjectClient={[Function]} inline={true}>
  <Tree className=\\"inline object-inspector\\" autoExpandAll={true} autoExpandDepth={0} disabledFocus={[undefined]} itemHeight={20} isExpanded={[Function]} isExpandable={[Function]} focused={{...}} getRoots={[Function]} getParent={[Function]} getChildren={[Function]} getKey={[Function]} onExpand={[Function]} onCollapse={[Function]} onFocus={[Function]} renderItem={[Function]}>
    <div className=\\"tree inline object-inspector\\" role=\\"tree\\" tabIndex=\\"0\\" onKeyDown={[Function]} onKeyPress={[Function]} onKeyUp={[Function]} onFocus={[Function]} onBlur={[Function]} onClick={[Function]} aria-label={[undefined]} aria-labelledby={[undefined]} aria-activedescendant={{...}} style={{...}}>
      <TreeNode id=\\"root\\" index={0} item={{...}} depth={0} renderItem={[Function]} focused={false} expanded={false} isExpandable={false} onExpand={[Function]} onCollapse={[Function]} onClick={[Function]}>
        <div id=\\"root\\" className=\\"tree-node\\" style={{...}} onClick={[Function]} role=\\"treeitem\\" aria-level={0} aria-expanded={[undefined]} data-expandable={false}>
          <div className=\\"node object-node\\" onClick={[Function]} onDoubleClick={{...}}>
            <span className=\\"object-label\\" onClick={{...}}>
              root
            </span>
            <span className=\\"object-delimiter\\">
              : 
            </span>
            <span className=\\"objectBox objectBox-number\\">
              42
            </span>
          </div>
        </div>
      </TreeNode>
    </div>
  </Tree>
</ObjectInspector>"
`;

exports[`ObjectInspector - classnames has the nowrap class when disableWrap prop is true 1`] = `
"<ObjectInspector autoExpandDepth={0} roots={{...}} createObjectClient={[Function]} disableWrap={true}>
  <Tree className=\\"nowrap object-inspector\\" autoExpandAll={true} autoExpandDepth={0} disabledFocus={[undefined]} itemHeight={20} isExpanded={[Function]} isExpandable={[Function]} focused={{...}} getRoots={[Function]} getParent={[Function]} getChildren={[Function]} getKey={[Function]} onExpand={[Function]} onCollapse={[Function]} onFocus={[Function]} renderItem={[Function]}>
    <div className=\\"tree nowrap object-inspector\\" role=\\"tree\\" tabIndex=\\"0\\" onKeyDown={[Function]} onKeyPress={[Function]} onKeyUp={[Function]} onFocus={[Function]} onBlur={[Function]} onClick={[Function]} aria-label={[undefined]} aria-labelledby={[undefined]} aria-activedescendant={{...}} style={{...}}>
      <TreeNode id=\\"root\\" index={0} item={{...}} depth={0} renderItem={[Function]} focused={false} expanded={false} isExpandable={false} onExpand={[Function]} onCollapse={[Function]} onClick={[Function]}>
        <div id=\\"root\\" className=\\"tree-node\\" style={{...}} onClick={[Function]} role=\\"treeitem\\" aria-level={0} aria-expanded={[undefined]} data-expandable={false}>
          <div className=\\"node object-node\\" onClick={[Function]} onDoubleClick={{...}}>
            <span className=\\"object-label\\" onClick={{...}}>
              root
            </span>
            <span className=\\"object-delimiter\\">
              : 
            </span>
            <span className=\\"objectBox objectBox-number\\">
              42
            </span>
          </div>
        </div>
      </TreeNode>
    </div>
  </Tree>
</ObjectInspector>"
`;
